### ISSUE TITLE
Support fast healing/regeneration

### ISSUE THREAD

**Issue by JamzTheMan** [CarlosFdez/pf2e-persistent-damage#7]
I know you named the module pf2e-persistent-damage but... :)

I modified the original macros to support this and it seemed to work well. Its basically the same thing except it happens at the beginning of the round and heals instead of damages the target.

It had handle things that stop regeneration so i created an "Anti-Regen" effect that i drag to the token if they did, say for instance, fire or acid damage to a troll, and only heal if that Effect was not on the target, and then removed the effect at the end of the round.

**Comment by CarlosFdez** [comment]
:)

I'll think about it but trying to handle the greater gamut of all timed events like afflictions/regen/etc is quite a massive scope. I still have quite a bit to do for persistent damage alone, including tracking sources of persistent damage somehow so that players can roll their own damage. Have to look into things like apply effect buttons on spell cards and all that fun stuff.

Currently I'm approaching the project like a proposal given that core seems to have a habit of absorbing modules (as far as what I've heard). Given that likely all of this will be handled in core eventually in some way or another (either taken as is, modified, or completely redone) I'd like to make sure its handled well so that a lot of repeated work isn't necessary.

So for example, how much does persistent damage share with afflictions, curses, and fast healing? They are probably different things entirely, but use the same supporting toolkit, so perhaps they would inherit some similar classes or lean into similar tools. Currently persistent damage is an effect with a special flag because new item types are a pain for modules to make, but how would it be handled in core? Would it be a new item type? A subtype of some existing type under a larger umbrella? How does this all fit in with conditions which have largely been left as a carcass on the roadside?

By focusing on the most annoying and most common one (persistent) we can begin with a small scope, but as soon as we get to all special conditions it becomes a much more difficult subject to talk about as we're essentially remaking the condition system.

So in other words, sure, but would need to think of an attack plan and would really love it if a core system dev chimed in somehow :(

**Comment by JamzTheMan** [comment]
Makes sense. It was purely selfish because I currently use a modified version of the same macro you started with but changed it so you could just enter a negative amount for fast healing and it pretty much worked.

Added a few more tweaks but it was pretty much akin to harm vs heal.

But i get it as well, I'd rather see your work absorbed into core as well, and if this would prevent that, I totally get it.

**Comment by CarlosFdez** [comment]
I mean, I'll be thinking about it (the added scope is scary though), I just wanna be sure I do it right with core in mind. I started writing down thoughts in #8 but its basically something I plan to edit as ideas come up.

**Comment by MSAbaddon** [comment]
Hi there,
as an idea since you already implemented a basic version of the healing rule element. As far as I can tell the blocking damage types in the rule element currently don't have any effect. Would be great if you add them to the healing chat card as a quick reminder.

**Comment by JamzTheMan** [comment]
@CarlosFdez I was testing Troll Regen and it seems to work ok (woot!) but the Suppress Regeneration does not seem to be working right now and seeing a console error (something about firstGM). It very well could be module conflict and need to do more testing on my end.

Is there any known issues with the regen/healing rules? (I notice PF2E throws a red warning in the Rules tab about unknown Rule Element)

**Comment by CarlosFdez** [comment]
It could be a regression due to core changes. I didn't see MSAbaddon's post but I'm suspecting that's the cause. I'll look into it soon, but secrets of magic stuff has me busy.

**Comment by CarlosFdez** [comment]
Late reply, but it was fixed a few days ago in a recent update. I'll close this ticket soon if there's nothing else.
